storage.ml

```diff
@@ -42,7 +44,11 @@ let t_of_json json =
                 let* user =
                   if v = 1 then User_model.user_v1_of_json js
                   else if v = 2 || v = 3 then User_model.user_v2_of_json js
-                  else User_model.user_of_json js
+                  else if v = 4 then
+                    User_model.(user_of_json cookie_v1_of_json) js
+                    (* TODO: the latest version which is 5 should use cookie_of_json but
+                       as of now it breaks functionality*)
+                  else User_model.(user_of_json cookie_v1_of_json) js
```

Comment:
@hannesm Because of this, the user-agent string only displays for the active session and all others display a placeholder string.
There is a `cookie_of_json` which I think should work with the latest version 5 but this is not the case.

Reply:
What is the issue in `cookie_of_json`?

Reply:
I went ahead in 3ae62de to use the `cookie_of_json` here. I think from previous version there were some cases missing: the last_access, user_agent, and uuid all being optional (would require 6 cases). it is simplified now, and should work [tm] ;)